src/plugins/unified_histogram/public/chart/use_time_range.tsx

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiIconTip, EuiText, useEuiTheme } from '@elastic/eui';
+import { css } from '@emotion/react';
+import type { IUiSettingsClient } from '@kbn/core-ui-settings-browser';
+import { i18n } from '@kbn/i18n';
+import React, { useCallback, useMemo } from 'react';
+import dateMath from '@kbn/datemath';
+import type { TimeRange } from '@kbn/data-plugin/common';
+import type { UnifiedHistogramBucketInterval } from '../types';
+
+export const useTimeRange = ({
+  uiSettings,
+  bucketInterval,
+  timeRange: { from, to },
+  timeInterval,
+}: {
+  uiSettings: IUiSettingsClient;
+  bucketInterval?: UnifiedHistogramBucketInterval;
+  timeRange: TimeRange;
+  timeInterval?: string;
+}) => {
+  const dateFormat = useMemo(() => uiSettings.get('dateFormat'), [uiSettings]);
+
+  const toMoment = useCallback(
+    (datetime?: moment.Moment) => {
+      if (!datetime) {
+        return '';
+      }
+      if (!dateFormat) {
+        return String(datetime);
+      }
+      return datetime.format(dateFormat);
+    },
+    [dateFormat]
+  );
+
+  const timeRangeText = useMemo(() => {
+    const timeRange = {
+      from: dateMath.parse(from),
+      to: dateMath.parse(to, { roundUp: true }),
+    };
+
+    const intervalText = i18n.translate('unifiedHistogram.histogramTimeRangeIntervalDescription', {
+      defaultMessage: '(interval: {value})',
+      values: {
+        value: `${
+          timeInterval === 'auto'
+            ? `${i18n.translate('unifiedHistogram.histogramTimeRangeIntervalAuto', {
+                defaultMessage: 'Auto',
+              })} - `
+            : ''
+        }${
+          bucketInterval?.description ??
+          i18n.translate('unifiedHistogram.histogramTimeRangeIntervalLoading', {
+            defaultMessage: 'loading...',
```

Comment:
Should this be init capped:  Loading

Reply:
I'm not sure, but definitely willing to update if so. It's being used in the time range display below the histogram until the chart is loaded, when it's replaced with the actual time interval returned from the request.

Current lowercase:
<img width="1023" alt="lowercase" src="https://user-images.githubusercontent.com/25592674/203883066-2917911b-f7b7-4329-b5a4-c130d2b2eec9.png">

Uppercase:
<img width="1023" alt="uppercase" src="https://user-images.githubusercontent.com/25592674/203883082-a8d43c4e-146f-4164-bc21-584ab7ccfb12.png">

WDYT?

Reply:
Uppercase

Reply:
Updated